lib/pavilion/cmd_utils.py

```diff
@@ -380,35 +386,23 @@ def build_local(tests: List[TestRun],
 
         test_threads = [thr for thr in test_threads if thr is not None]
 
-        if cancel_event.is_set():
-            for thread in test_threads:
-                thread.join()
+        # Hard-Fail. One build fails, all other builds are aborted.
+        if len(tests) == 1:
+            hard_fail = True
+        for build, event in fail_events.items():
+            if event.is_set() and hard_fail:
+                cancel_event.set()
+                for thread in test_threads:
+                    thread.join()
 
-            for test in tests:
-                if (test.status.current().state not in
-                        (STATES.BUILD_FAILED, STATES.BUILD_ERROR)):
-                    test.status.set(
-                        STATES.ABORTED,
-                        "Run aborted due to failures in other builds.")
-
-            output.fprint(
-                "Build error while building tests. Cancelling runs.",
-                color=output.RED, file=outfile, clear=True)
-            output.fprint(
-                "Failed builds are placed in <working_dir>/test_runs/"
-                "<test_id>/build for the corresponding test run.",
-                color=output.CYAN, file=outfile)
-
-            for failed_build in mb_tracker.failures():
-                output.fprint(
-                    "Build error for test {f.test.name} (#{f.test.id})."
-                    .format(f=failed_build), file=errfile)
-                output.fprint(
-                    "See test status file (pav cat {id} status) and/or "
-                    "the test build log (pav log build {id})"
-                    .format(id=failed_build.test.id), file=errfile)
-
-            return errno.EINVAL
+                for test in tests:
+                    if (test.status.current().state not in
+                            (STATES.BUILD_FAILED, STATES.BUILD_ERROR, STATES.BUILD_TIMEOUT)):
+                        test.status.set(
+                            STATES.ABORTED,
+                            "Run aborted due to failures in other builds.")
```

Comment:
Cancel based on build hash, not status file.

Reply:
This doesn't actually cancel tests, it sets the proper `ABORTED` status for test/threads that don't get started prior to the cancel event. That being said, the only status I need to actually check for here is `STATE.CREATED` and I did that.